File: subprojects/java-compatibility/src/test/java/kotools/types/number/NonZeroIntTest.java
package kotools.types.number;

import kotools.types.internal.ErrorMessage;
import kotools.types.internal.ErrorMessageKt;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import java.util.Random;

public class NonZeroIntTest {
    @Nested
    public class Companion {
        @Test
        public void create_should_pass_with_a_Number_other_than_zero() {
            final Random random = new Random();
            int value;
            do {
                value = random.nextInt();
            } while (value == 0);
            final Number number = value;
            Assertions.assertDoesNotThrow(
                    () -> NonZeroInt.Companion.create(number)
            );
        }

        @Test
        public void create_should_fail_with_a_Number_that_equals_zero() {
            final IllegalArgumentException exception = Assertions.assertThrows(
                    IllegalArgumentException.class,
                    () -> NonZeroInt.Companion.create(0)
            );
            final ErrorMessage actual = ErrorMessageKt.ErrorMessage(exception);
            final ErrorMessage expected =
                    ErrorMessage.Companion.getZeroNumber();
            Assertions.assertEquals(expected, actual);
        }

        @Test
        public void createOrNull_should_pass_with_a_Number_other_than_zero() {
            final Random random = new Random();
            int value;
            do {
                value = random.nextInt();
            } while (value == 0);
            final Number number = value;
            final NonZeroInt actual = NonZeroInt.Companion.createOrNull(number);
            Assertions.assertNotNull(actual);
        }

        @Test
        public void createOrNull_should_fail_with_a_Number_that_equals_zero() {
            final NonZeroInt actual = NonZeroInt.Companion.createOrNull(0);
            Assertions.assertNull(actual);
        }
    }
}

File: src/test/kotlin/io/github/kotools/types/int/NonZeroIntTest.kt
package io.github.kotools.types.int

import io.github.kotools.assert.assertEquals
import io.github.kotools.assert.assertFails
import io.github.kotools.assert.assertNotNull
import io.github.kotools.assert.assertNull
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.assertDoesNotThrow
import kotlin.test.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class NonZeroIntTest {
    @Nested
    inner class CompareTo {
        @Test
        fun `should pass`(): Unit = assertTrue { 1.nonZero < 2.nonZero }

        @Test
        fun `should fail`(): Unit = assertFalse { 1.nonZero > 2.nonZero }
    }

    @Nested
    inner class Create {
        @Test
        fun `should pass`(): Unit = assertDoesNotThrow { NonZeroInt create 1 }

        @Test
        fun `should fail`(): Unit = assertFails(NonZeroInt.ERROR_MESSAGE) {
            NonZeroInt create 0
        }
    }

    @Nested
    inner class CreateOrNull {
        @Test
        fun `should pass`() {
            NonZeroInt.createOrNull(1)
                .assertNotNull()
        }

        @Test
        fun `should fail`(): Unit = NonZeroInt.createOrNull(0)
            .assertNull(NonZeroInt.ERROR_MESSAGE)
    }

    @Nested
    inner class Div {
        @Test
        fun `should pass`(): Unit =
            (6.nonZero / 3.nonZero) assertEquals 2.nonZero
    }

    @Nested
    inner class Times {
        @Test
        fun `should pass`(): Unit =
            (2.nonZero * 4.nonZero) assertEquals 8.nonZero
    }

    @Nested
    inner class UnaryMinus {
        @Test
        fun `should pass`(): Unit = -(1.nonZero) assertEquals (-1).nonZero
    }
}
